sources/packages/backend/apps/api/src/route-controllers/user/user.institutions.controller.ts

```diff
@@ -0,0 +1,128 @@
+import { BCeIDAccountTypeCodes } from "../../services/bceid/bceid.models";
+import { UserControllerService } from "..";
+import { ClientTypeBaseRoute } from "../../types";
+
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@Controller("users")
+@ApiTags(`${ClientTypeBaseRoute.Institution}-user`)
+export class UserInstitutionsController extends BaseController {
+  constructor(
+    private readonly userService: UserService,
+    private readonly bceidService: BCeIDService,
+    private readonly userControllerService: UserControllerService,
+  ) {
+    super();
+  }
+
+  /**
+   * Get the BCeID account information from BCeID Web services
+   * for the currently authenticated user.
+   * @param userToken authenticated user token.
+   * @returns BCeID account information from BCeID Web services.
+   */
+  @AllowInactiveUser()
+  @Get("bceid-account")
+  async getBCeID(
+    @UserToken() userToken: IUserToken,
+  ): Promise<BCeIDDetailsAPIOutDTO | null> {
+    const account = await this.bceidService.getAccountDetails(
+      userToken.idp_user_name,
+      BCeIDAccountTypeCodes.Business,
+    );
+    if (account == null) {
+      return null;
+    } else {
+      return {
+        user: {
+          guid: account.user.guid,
+          displayName: account.user.displayName,
+          firstname: account.user.firstname,
+          surname: account.user.surname,
+          email: account.user.email,
+        },
+        institution: {
+          guid: account.institution.guid,
+          legalName: account.institution.legalName,
+        },
+      };
+    }
+  }
+
+  /**
+   * Gets all business BCeID accounts information for the
+   * for the currently authenticated user.
+   * @param userToken authenticated user token.
+   * @returns all business BCeID accounts.
+   */
+  @ApiUnprocessableEntityResponse({
+    description:
+      "Not able to retrieve BCeID business account details for the current authenticated user.",
+  })
+  @Get("bceid-accounts")
+  async getAllBCeIDs(
+    @UserToken() userToken: IUserToken,
+  ): Promise<BCeIDAccountsAPIOutDTO> {
+    return await this.userControllerService.getAllBCeIDs(userToken);
+  }
+
+  /**
+   * Gets the user details for the authenticated institution user.
+   * @param userToken authenticated user token.
+   * @returns institution user details.
+   */
+  @ApiUnprocessableEntityResponse({
+    description: "No user record found for user.",
+  })
+  @Get()
+  async institutionDetail(
+    @UserToken() userToken: IUserToken,
+  ): Promise<InstitutionUserAPIOutDTO> {
+    const user = await this.userService.getActiveUser(userToken.userName);
+    if (!user) {
+      throw new UnprocessableEntityException("No user record found for user.");
+    }
+    return {
+      userEmail: user.email,
+      userFirstName: user.firstName,
+      userLastName: user.lastName,
+    };
+  }
+
+  /**
+   * Updates the institution user details.
+   * @param userToken authenticated user token.
```

Comment:
Just minor, the `userToken` we usually suppress from the comments because it is the same for every situation.